identity/app/clients/DiscussionClient.scala

```diff
@@ -82,4 +94,29 @@ class DiscussionClient(wsClient: WSClient, config: conf.IdentityConfiguration)(i
       case e => Future.failed(new DiscussionApiServiceException(s"findProfileStats failure to find $userId", e))
     }
   }
+
+  /**
+    * Check if the user has at least one comment. This is a lightweight check which replaces
+    * the findProfileStats call, which has been found to be extremely slow for users with a large
+    * number of comments, as on the Discussion API side it involves a full count of all comments.
+    * This method simply requests a single comment, and checks if the response is non-empty.
+    * If it is, the user has at least one publicly visible comment, and their profile can be shown.
+    */
+  def profileHasAtLeastOneComment(userId: String): Future[Boolean] = {
+
+    val apiPath = s"/profile/$userId/comments?page=1&pageSize=1"
+
+    val discussionCommentsF = for {
+      response <- GET(apiPath)
+      userCommentsResponse <- Future.fromTry(handleApiResponse(response, userId))
+    } yield userCommentsResponse.map(_.as[DiscussionCommentsResponse].comments.nonEmpty)
+
+    discussionCommentsF.recoverWith {
+      case e =>
+        Future.failed(new DiscussionApiServiceException(s"profileHasAtLeastOneComment failure to find $userId", e))
+    }
```

Comment:
Could be simplified as:
```suggestion
    discussionCommentsF.recover {
      case e =>
        new DiscussionApiServiceException(s"profileHasAtLeastOneComment failure to find $userId", e)
    }
```

Reply:
Great - I've combined yours and Patricio's suggestions below.